assets/js/googlesitekit/api/cache.test.js

```diff
@@ -2,9 +2,40 @@
  * Internal dependencies
  */
 // eslint-disable-next-line @wordpress/dependency-group
-import { _setStorageKeyPrefix, _setSelectedStorageBackend, get, set, deleteItem, getKeys, clearCache } from './cache';
+import { _setStorageKeyPrefix, _setSelectedStorageBackend, _getStorage, get, set, deleteItem, getKeys, clearCache } from './cache';
+
+let previousCacheValue;
+const disableCache = () => {
+	previousCacheValue = global.googlesitekit.admin.nojscache;
+	global.googlesitekit.admin.nojscache = true;
+};
+
+const restoreCache = () => {
+	global.googlesitekit.admin.nojscache = previousCacheValue;
+};
+
+const DISABLE_CACHE = 'Cache disabled';
+const NO_BACKEND = 'Null backend';
 
 describe( 'googlesitekit.api.cache', () => {
+	describe( '_getStorage', () => {
+		it( 'should return the most applicable storage driver available', async () => {
+			const storage = await _getStorage();
+
+			// localStorage is the best storage mechanism available in the test suite
+			// and should be returned.
+			expect( storage ).toEqual( localStorage );
```

Comment:
In our case "best" really means the first available in the list of `storageOrder` backends. This test might be a bit more clear if the order was something like `[ 'unavailableBackend', 'localStorage', 'sessionStorage' ]` or do you feel like that would be getting too concerned with implementation details?

Reply:
I'm cool with that. I'll refactor these functions a bit so it's easier to set that up and test it.